pcsx2/CDVD/CsoFileReader.cpp

```diff
@@ -256,18 +264,34 @@ int CsoFileReader::ReadChunk(void* dst, s64 chunkID)
 		// This might be less bytes than frameRawSize in case of padding on the last frame.
 		// This is because the index positions must be aligned.
 		const u32 readRawBytes = fread(m_readBuffer.get(), 1, frameRawSize, m_src);
+		bool success = false;
 
-		m_z_stream->next_in = m_readBuffer.get();
-		m_z_stream->avail_in = readRawBytes;
-		m_z_stream->next_out = static_cast<Bytef*>(dst);
-		m_z_stream->avail_out = m_frameSize;
+		if (m_uselz4)
+		{
+			int src_size = static_cast<int>(readRawBytes);
+			int dst_size = static_cast<int>(m_frameSize);
+			const char* src_buf = reinterpret_cast<const char*>(m_readBuffer.get());
+			char* dst_buf = static_cast<char*>(dst);
+			
+			const int res = LZ4_decompress_safe_partial(src_buf, dst_buf, src_size, dst_size, dst_size);
+			success = res > 0;
+		}
+		else
+		{
+			m_z_stream->next_in = m_readBuffer.get();
+			m_z_stream->avail_in = readRawBytes;
+			m_z_stream->next_out = static_cast<Bytef*>(dst);
+			m_z_stream->avail_out = m_frameSize;
 
-		int status = inflate(m_z_stream.get(), Z_FINISH);
-		bool success = status == Z_STREAM_END && m_z_stream->total_out == m_frameSize;
+			const int status = inflate(m_z_stream.get(), Z_FINISH);
+			success = status == Z_STREAM_END && m_z_stream->total_out == m_frameSize;
+		}
 
 		if (!success)
-			Console.Error("Unable to decompress CSO frame using zlib.");
-		inflateReset(m_z_stream.get());
+			Console.Error(fmt::format("Unable to decompress CSO frame using {}", ((m_uselz4)? "lz4":"zlib")));
```

Comment:
```suggestion
			Console.Error(fmt::format("Unable to decompress CSO frame using {}", m_uselz4 ? "lz4" : "zlib"));
```
Parens aren't needed here.